### PR TITLE
Refactoring rounds db repository - Closes #1393

### LINKED ISSUE #1393: Refactor rounds db repository
Parent #1376

### PR DESCRIPTION
**Pull request by vitaly-t**
Closes #1393 

This one was branched from the branch for PR 1381. That's why it needs to be updated after that PR 1381 has been merged in, and only then it can be reviewed and merged.

---

**IMPORTANT:** This PR turns out to be **URGENT**. It was branched out from the branch of PR 1381, which somehow missed one important commit during its post-review changes, this line: https://github.com/LiskHQ/lisk/blob/development/db/sql/blocks/getIdSequence.sql#L21

`${LIMIT}` is supposed to be lower-case, or else the query won't work. And this PR also rectifies that.

### CODE REVIEW

File: db/sql/index.js [hunk 1]
```diff
@@ -41,6 +41,20 @@ module.exports = {
...
+		summedRound: load('rounds/summedRound.sql'),
+		clearRoundSnapshot: load('rounds/clearRoundSnapshot.sql'),
+		performRoundSnapshot: load('rounds/performRoundSnapshot.sql'),
+		restoreRoundSnapshot: load('rounds/restoreRoundSnapshot.sql'),
+		clearVotesSnapshot: load('rounds/clearVotesSnapshot.sql'),
+		performVotesSnapshot: load('rounds/performVotesSnapshot.sql'),
+		restoreVotesSnapshot: load('rounds/restoreVotesSnapshot.sql'),
+		getMemRounds: load('rounds/getMemRounds.sql'),
+		flush: load('rounds/flush.sql'),
+		truncateBlocks: load('rounds/truncateBlocks.sql')
```
Comment (nazarhussain) [inline db/sql/index.js]: @vitaly-t Everything looks good to me. Later we may need to introduce some auto loading pattern. So we just need to create an SQL file and just use it with its name in db repos. May be using `require-all` we can do it.

File: db/sql/blocks/getIdSequence.sql [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 WITH current_round AS
   (
-    SELECT CEIL(b.height / ${delegates}::float)::bigint
+    SELECT ceil(b.height / ${delegates}::float)::bigint
```
Comment (nazarhussain) [inline db/sql/blocks/getIdSequence.sql]: Although it works, is there any specific reason to change `CEIL` to down case?

Comment (vitaly-t) [reply db/sql/blocks/getIdSequence.sql]: Actually, I changed all standard SQL functions to low case, not just here. The convention suggests - you use upper case only for SQL key words + CONSTANTS, but not for function names :wink:

File: db/sql/index.js (hunk 1 above)
Comment (vitaly-t) [reply db/sql/index.js]: NOOO 😸 That's completely in the wrong direction. Those are all dynamic file links, they are not just some SQL-loaded content. They are self-debugging, self-updating, self-error-reporting, etc. `require-all` would kill all that :smile:

File: db/sql/blocks/getIdSequence.sql [hunk 3]
```diff
@@ -18,7 +18,7 @@ WITH current_round AS
                          (SELECT *
                           FROM current_round),
                           (SELECT *
-                           FROM current_round) - ${LIMIT} + 1, -1))
+                           FROM current_round) - ${limit} + 1, -1))
```
Comment (4miners) [inline db/sql/blocks/getIdSequence.sql]: 👍

File: db/sql/blocks/getIdSequence.sql (hunk 2 above)
Comment (nazarhussain) [reply db/sql/blocks/getIdSequence.sql]: 👍

**Pull request merged**